sycl/source/backend.cpp

```diff
@@ -276,6 +285,19 @@ kernel make_kernel(const context &TargetContext,
         *KernelBundle.begin();
     const auto &DeviceImageImpl = getSyclObjImpl(DeviceImage);
     PiProgram = DeviceImageImpl->get_program_ref();
+  } else if (Backend == backend::opencl) {
+    cl_kernel CLKernel = detail::pi::cast<cl_kernel>(NativeHandle);
+    cl_program CLProgram;
+    size_t Ret = clGetKernelInfo(CLKernel, CL_KERNEL_PROGRAM, sizeof(CLProgram),
+                                 &CLProgram, nullptr);
+    if (Ret != CL_SUCCESS) {
+      throw runtime_error(
+          "Failed to retrieve program associated with the kernel",
+          PI_ERROR_INVALID_KERNEL);
+    }
+    Plugin->call<detail::PiApiKind::piextProgramCreateWithNativeHandle>(
+        detail::pi::cast<pi_native_handle>(CLProgram),
+        ContextImpl->getHandleRef(), false, &PiProgram);
```

Comment:
Some sycl interop apis like `make_kernel` could be called without passing the dependency handles with it like the program associated with the kernel and at the same time UR entry `urKernelCreateWithNativeHandle` need the UR program handle as a mandatory requirement, we need to query it from opencl and create a new object in UR layer for the program that will be saved to be associated with the new created kernel object. This is also true for other interop apis so this was also handled like that for other types of handles.